runtime/src/iree/hal/drivers/webgpu/platform/emscripten/emscripten_util.c

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/hal/drivers/webgpu/platform/webgpu.h"
+
+//===----------------------------------------------------------------------===//
+// Implementation compatibility layer
+//===----------------------------------------------------------------------===//
+
+#define WGPU_EMSCRIPTEN_INSTANCE ((WGPUInstance)((uintptr_t)0xABADD00Du))
+
+WGPUInstance wgpuCreateInstance(WGPUInstanceDescriptor const* descriptor) {
+  // Emscripten does not have instances (yet?)
+  // We use a sentinel value here so that we can do null checks in places for
+  // implementations that do use instances.
+  return WGPU_EMSCRIPTEN_INSTANCE;
+}
+
+void iree_wgpuBindGroupDrop(WGPUBindGroup bindGroup) {
```

Comment:
do the latest headers have the destroys for some of these? I think there's a buffer and device destroy now we can use (https://www.w3.org/TR/webgpu/#dom-gpudevice-destroy + https://www.w3.org/TR/webgpu/#dom-gpubuffer-destroy)

Reply:
For some (buffer, texture, queryset, device) yes.

* https://github.com/emscripten-core/emscripten/blob/704218f6a31dd5d2d584c5ec117b09ca424ffd0e/src/library_webgpu.js#L672-L693
* https://github.com/webgpu-native/webgpu-headers/blob/main/webgpu.h

Reply:
Cool - we can clean those up later - not sure what we're supposed to do with other things though (still a massive oversight for native webgpu IMO, but I'm not sure how it doesn't leak with emscripten either!)